pkg/webhook/podhandler/podhandler.go

```diff
@@ -125,5 +126,40 @@ func (p *PodHandler) Handle(ctx context.Context, req admission.Request) admissio
 		return admission.Errored(http.StatusInternalServerError, err)
 	}
 
-	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
+	rv := admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
+	return rv
+}
+
+func (p *PodHandler) podHandlerHelper(podToModify *corev1.Pod, sideCars []corev1.Container, volumes []corev1.Volume, volumeMounts []corev1.VolumeMount) *admission.Response {
+	duplicateValuesMsg := "webhook mutation would result in duplicate values, returning"
+	for idx, sideCar := range sideCars {
+		sideCar := sideCar
+		if seqs.Any(seqs.FromSlice(podToModify.Spec.Containers), func(c corev1.Container) bool {
+			return c.Name == sideCar.Name
+		}) {
+			p.Log.Info(duplicateValuesMsg)
+			rv := admission.Allowed(duplicateValuesMsg)
```

Comment:
Why not admission.Denied? I beleive this should result in an error on the user side

Reply:
My reasoning was that it would not be idempotent if we denied the request, but after giving it a bit more thought it seems to be idempotent on the server side (appends only if needed) but the user indeed should receive an error.